Mage/Module.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+
+using Newtonsoft.Json.Linq;
+
+
+public class Module<T> : Singleton<T> where T : class, new() {
+	//
+	protected Mage mage { get { return Mage.instance; } }
+	protected Logger logger { get { return mage.logger(this.GetType().Name); } }
```

Comment:
it is a singleton per logger type, idea, once instantiated for type "User" it will retrieve the same instance
check https://github.com/mage/mage-sdk-unity/pull/13/files#diff-7dbe7788b73f1c30c06e744c30d95e85R28